Refactor code, add a lot of documentation

plainerman opened the pull request.

plainerman committed:
Refactor code, add documentation

prajjwaldimri commented on Genius/ContentPoster.cs:
```diff
@@ -3,25 +3,20 @@
-        public static string AuthorizationToken { get; set; }
-
+	    /// <summary>
+	    /// The token that is used for authorization.
```
No need to state the obvious in comments. AuthorizationToken self explains what it is

prajjwaldimri commented on Genius/ContentPoster.cs:
```diff
@@ -122,12 +117,24 @@ public class ContentPoster
         [JsonProperty(PropertyName = "referent")]
         public Referent Referent { get; set; }
+		/// <summary>
+		/// The web page.
```
Again, No need to comment webpage on a variable named WebPage. It self-explains itself.

prajjwaldimri commented on Genius/Authenticator.cs:
```diff
@@ -9,170 +9,161 @@
 
 namespace Genius
 {
```
The whole file is shown as changed however nothing is 'actually changed' in the file. Please revert this file to the latest commit so that merging is clean.

prajjwaldimri commented on Genius/Voter.cs:
```diff
@@ -11,24 +11,35 @@ namespace Genius
     {
         /// <summary>
-        /// 
+        /// The token that is used for authorization.
```
No need for comments on self-explaining variables.

prajjwaldimri commented on Genius/Voter.cs:
```diff
@@ -11,24 +11,35 @@ namespace Genius
+		/// <param name="annotationId">Id for the annotation to be voted on</param>
+		/// <param name="vote"></param>
+		/// <returns></returns>
+		private static async Task Vote(string annotationId, string vote)
```
Probably use an enum for vote string? Would be cleaner.

prajjwaldimri commented on Genius/Search.cs:
```diff
@@ -36,10 +47,22 @@ public class Search
     public class Hit
     {
+		/// <summary>
+		/// The index of this hit.
```
Unnecessary Comment

prajjwaldimri commented on Genius/Search.cs:
```diff
@@ -3,23 +3,34 @@
-        public static string SearchTerm { get; set; }
+	    /// <summary>
+	    /// The token that is used for authorization.
+	    /// </summary>
```
Unnecessary Comment

prajjwaldimri commented on Genius/Search.cs:
```diff
@@ -3,23 +3,34 @@
 {
+	/// <summary>
+	/// Ability to perform a search.
+	/// </summary>
```
Unnecessary Comment

prajjwaldimri commented on Genius/Data/Referent.cs:
```diff
@@ -1,24 +1,33 @@
 
+		/// <summary>
+		/// The id of the annotator.
+		/// </summary>
```
Unnecessary Comment

prajjwaldimri commented on Genius/Data/Referent.cs:
```diff
@@ -1,24 +1,33 @@
     {
+		/// <summary>
+		/// The type of the referent.
+		/// </summary>
```
Unnecessary Comment

prajjwaldimri commented on Genius/Data/Media.cs:
```diff
@@ -1,6 +1,6 @@
+﻿namespace Genius.Data
 {
-    public class Media
+	public class Media
```
Weird spacing?

prajjwaldimri commented on Genius/Data/Author.cs:
```diff
@@ -0,0 +1,21 @@
+	public class Author
+    {
+		/// <summary>
+		/// The attribution.
```
Unnecessary Comment

prajjwaldimri commented on Genius/Data/Author.cs:
```diff
@@ -0,0 +1,21 @@
+		/// </summary>
+        public string Attribution { get; set; }
+		/// <summary>
+		/// The pinned role of the author.
```
Unnecessary Comment

prajjwaldimri commented:
@plainerman Thanks for submitting the pull-request :) Please take a look at the suggested changes.

plainerman committed:
Update to API Version 2.0

Fix errors with new Genuis API
Converted the API to be object-based instead of statics
+ Ability for multiple threads to use API
+ Easier to extend and adapt

plainerman committed:
Fix weird spacing in media class

plainerman commented:
I converted the API to be object-based, which resolved a lot of issues I had with the API.

1. Objects can be accessed from multiple threads at the same time / support for multiple authentication tokens at the same time
2. A lot less code duplication (new requests only require a line or two)

Further, the Genius getSong-API changed - I documented it in Annotation.cs, maybe you know how to fix it? (I uncommented the lines that caused a parsing exception)

The possible weird spacing is due to the difference of tabs and spaces, now everything should be tab-based.

Before merging, the Wiki and README.md has to be updated.

prajjwaldimri commented:
@plainerman Sorry for the late reply. I just returned from vacation today. Just updating you that I will take a look at the PR tomorrow.

prajjwaldimri commented:
Closing this PR as I am completely rewriting the library. You can see the progress and contribute to the development on the [develop](https://github.com/prajjwaldimri/Genius.NET/tree/develop) branch or  [![Join the chat at https://gitter.im/Genius-NET/Lobby](https://badges.gitter.im/Genius-NET/Lobby.svg)](https://gitter.im/Genius-NET/Lobby?utm_source=badge&utm_medium=badge&utm_campaign=pr-badge&utm_content=badge)
